lib/resty/dns/resolver.lua

```diff
@@ -832,6 +839,69 @@ end
 
 
 function _M.query(self, qname, opts, tries)
+    if self.doh then
+        return _M.doh_query(self,qname,opts,tries)
+    end
+
+    return _M.udp_tcp_query(self,qname,opts,tries)
+end
+
+function _M.doh_query(self, qname, opts, tries)
+    local retrans = self.retrans
+    if tries then
+        tries[1] = nil
+    end
+
+    local servers = self.servers
+
+    if #servers == 0 then
+        return nil, "No servers available"
+    end
+
+    local err
+
+    for i = 1, retrans do
+        local idx = i
+
+        if idx > #servers then
+            idx = 1
+        end
+
+        local res
+        local id 
+        
+        if self.doh == 'GET' then
+            res = ngx.location.capture(servers[idx][1] .. b64.encode_base64url(qname))
+        else
+            id = _gen_id(self)
+            local bdata = table.concat(_build_request(qname, id, self.no_recurse, opts))
+            res = ngx.location.capture(servers[idx][1],{ method = ngx.HTTP_POST, body = bdata })
+        end
+
+        if res.status == 200 and res.body then
```

Comment:
maybe we should check if `res` is not nil here

Reply:
done